packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -443,9 +443,13 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox {
 
   void _handleDragEnd(DragEndDetails details) {
     if (_position.value >= 0.5)
-      _positionController.forward();
+      _positionController.forward().whenComplete(() {
+        if (!_value) _positionController.reverse();
```

Comment:
Per our [style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#separate-the-if-expression-from-its-statement) :
> Don’t put the statement part of an 'if' statement on the same line as the expression, even if it is short.